api/scanner/scanner_media.go

```diff
@@ -10,6 +13,30 @@ import (
 	"github.com/viktorstrate/photoview/api/graphql/models"
 )
 
+func scanForSideCarFile(path string) (string, error) {
+	testPath := path + ".xmp"
+	_, err := os.Stat(testPath)
+	if os.IsNotExist(err) {
+		return "", err
+	} else {
+		return testPath, nil
+	}
```

Comment:
Again maybe return a `*string` and set it to nil instead of an empty string.

And then only return an error if the error is somthing other than the file doesn't exist, as then it might be worth logging the error.
See here as an example:
https://github.com/viktorstrate/photoview/blob/940e48fee3f320e60eed284c4c5f49688b011273/api/routes/spa.go#L43-L53

Reply:
I've amended it to return *string. Bit unsure about the logging, hopefully its formatted correctly.